plugins/modules/packaging/os/zypper_repository.py

```diff
@@ -186,6 +186,14 @@ def _repo_changes(realrepo, repocmp):
             valold = str(repocmp[k] or "")
             valnew = v or ""
             if k == "url":
+                cmd = ['rpm', '-q', '--qf', '%{version}', '-f', '/etc/os-release']
+                rc, stdout, stderr = module.run_command(cmd, check_rc=True)
+                valnew = valnew.replace('$releasever', stdout)
+                valold = valold.replace('$releasever', stdout)
+                cmd = ['rpm', '-q', '--qf', '%{arch}', '-f', '/etc/os-release']
+                rc, stdout, stderr = module.run_command(cmd, check_rc=True)
+                valnew = valnew.replace('$basearch', stdout)
+                valold = valold.replace('$basearch', stdout)
```

Comment:
How about only running these commands `if '$releasever' in valold or '$releasever' in vernew:` for the first, and similar for the second? Then people who don't use these variables do not get extra runtime for executing `rpm` for each of them.

Reply:
Nice idea. Added in https://github.com/ansible-collections/community.general/pull/2722/commits/b1db85c5043ecd9332d4b5c112d9cf6a9e67106f.